battle.cpp

```diff
@@ -57,7 +57,7 @@ bool IsArmyOverwhelmedBy(Army * a, Army * b) {
 		bPower = b->NumFrontHits();
 	}
 
-	return bPower > aPower && a->NumFront() && a->NumBehind();
+	return bPower > aPower;
```

Comment:
why this change? is there any issue with Overwhelming it fixes?

Reply:
redundant